docs/architecture/adr-015-improve-chain-link-signature-support.md

```diff
@@ -0,0 +1,147 @@
+message CosmosSignature {
+  // Encoding of the signed value 
+  ValueEncoding value_encoding = 1;
+  
+  // Data of the signature
+  google.protobuf.Any signature_data = 2;
+  
+  // ValueEncoding contains the different kinds of value encoding that the signature can have
+  enum ValueEncoding {
+    // Unknown should never be used
+    Unknown = 0;
+    
+    // TXLegacyAmino should be used when the signed value is an Amino transaction, containing the address inside its memo 
+    TXLegacyAmino = 1;
+    
+    // TXDirect should be used when the signed value is a Protobuf transaction containing the address inside its memo
+    TXDirect = 2;
+    
+    // Textual should be used when the signed value is the address itself
+    Textual = 3;
+  }
+}
+
+// CosmosSingleSignatureData contains the signature data of a single Cosmos-compatible signature
+message CosmosSingleSignatureData {
+  // Raw signature bytes
+  bytes signature = 1;
+}
+
+// CosmosMultiSignatureData contains the signature data of a Cosmos-compatible multisig
+message CosmosMultiSignatureData {
+  // Bitarray specifies which keys within the multisig are signing
+  cosmos.crypto.multisig.v1beta1.CompactBitArray bit_array = 1;
+
+  // Signatures of the multi-signature
+  repeated google.protobuf.Any signatures = 2;
+}
+```
+
+### `EVMSignature`
+
+The `EVMSignature` implementation of the `Signature` interface will be as follows:
+
+```protobuf
+syntax = "proto3";
+
+// EVMSignature contains the data of a signature generated using an EMV-compatible wallet
+message EVMSignature {
+  // Method used when generating the signature
+  Method signing_method = 1;
+  
+  // Raw signature bytes
+  bytes signature = 2;
+  
+  enum Method {
+    // Unknown should never be used
+    Unknown = 0;
+    
+    // PersonalSign should be used when the value has been signed using the personal_sign signature method
+    PersonalSign = 1;
+  }
+}
+```
+
+## Consequences
+
+### Backwards Compatibility
+
+The proposed solution is partially backward compatible, but a store migration is still required. Particularly, for each link:
+- all `SingleSignatureData` instances should be converted to `CosmosSignature` containing a `CosmosSingleSignatureData`; 
+- all `MultiSignatureData` instances should be converted to `CosmosSignature` containing a `CosmosMultiSignatureData`;
+
+Also, the `Proof#Verify` method must be updated in order to not check for different signature types, but to simply rely on the `Signature#Verify` method instead.
+
+### Positive
+
+- Allows for a greater extensibility in the future 
+- Makes it easier to verify a signature
+
+### Negative
+
+### Neutral
+
+## Further Discussions
+
+We should perform a search within the Solana ecosystem to understand what are the most used wallets, and what kind of signature specification they allow the user to use in order to sign arbitrary data. This will allow us to implement this kind of `Signature` as well. 
```

Comment:
I've done some researches on Solana. By looking at different DeFi/NFTs application in the ecosystem:
- [Saber](https://saber.so/)
- [Raydium](https://raydium.io/)
- [Mango](https://mango.markets/)
- [Step](https://app.step.finance)
- [Opensea](https://opensea.io/)

It appears that the most common wallets across these all platforms are, in order of popularity:
- Phantom (This is and Glow are the only ones supported by opensea)
- Solflare
- Sollet
- Glow

Here some info on the signatures:
 - Signature Format: `ed25519` binary consuming 64 bytes;
 - Message encoding: `hex` or `UTF-8`;
[This](https://docs.phantom.app/integrating/extension-and-mobile-browser/signing-a-message) is how signing arbitrary data is performed by Phantom.
More [here](https://codesandbox.io/s/github/phantom-labs/sandbox?file=/src/App.tsx). 

This is what I found looking for it, maybe @dadamu can clarify further